jena-fuseki2/jena-fuseki-main/src/main/java/org/apache/jena/fuseki/main/sys/FusekiModules.java

```diff
@@ -38,35 +40,45 @@ public static void load() {
     }
 
     public static void reload() {
-        registry = ConcurrentHashMap.newKeySet();
+        registry = new ArrayList<>();
         subsystem = new Subsystem<FusekiModule>(FusekiModule.class);
         subsystem.initialize();
-        subsystem.forEach(registry::add);
+        synchronized(lock) {
+            subsystem.forEach(registry::add);
+        }
     }
 
     /** Add a code module */
     public static void add(FusekiModule module) {
-        load();
-        module.start();
-        registry.add(module);
+        synchronized(lock) {
+            load();
+            module.start();
+            registry.add(module);
+        }
     }
 
     /** Remove a code module */
     public static void remove(FusekiModule module) {
-        registry.remove(module);
-        module.stop();
+        synchronized(lock) {
+            registry.remove(module);
+            module.stop();
+        }
     }
 
     /** Test whether a code module is registered. */
     public static boolean contains(FusekiModule module) {
-        return registry.contains(module);
+        synchronized(lock) {
+            return registry.contains(module);
+        }
     }
 
     /*package*/ static void forEachModule(Consumer<FusekiModule> action) {
-        if ( registry == null )
-            load();
-        if ( registry == null || registry.isEmpty() )
-            return ;
-        registry.forEach(action);
+        synchronized(lock) {
+            if ( registry == null )
+                load();
+            if ( registry == null || registry.isEmpty() )
```

Comment:
I think the `load()` call above means the `registry` cannot be null (it'll be at least an empty arraylist?)

Reply:
Yes - but safe code in case `load` changes can leave the registry as null. Null tests are free! Its sort of left over from before `synchonized`. Multithreading startup complexity now avoided.

Reply:
>Yes - but safe code in case load changes can leave the registry as null. Null tests are free! Its sort of left over from before synchonized. Multithreading startup complexity now avoided.

:+1: sounds good to me, thanks!